### PR TITLE
Add support for 11.0.0 and drop support for 8.0.0

### PR DESCRIPTION
**Pull request by klausenbusk**
#356 is currently blocked on this PR.

### CODE REVIEW

**Comment by klausenbusk** [comment]
I assume the error is caused by this change: https://github.com/keycloak/keycloak/commit/6005503a3dd7b65b79e8c3289738cb5d60779d8f
```
Error: error sending POST request to /auth/admin/realms/terraform-tduhynsuat/components: 400 Bad Request. Response body: {"errorMessage":"Missing configuration for Groups Path"}
```

**Comment by klausenbusk** [comment]
... and now I broke all  <11.0.0.. How can we support 11.0.0 without breaking < 11.0.0?

**Comment by mrparkers** [comment]
Yeah... I was afraid that something like this might happen due to a major release introducing breaking API changes.

I haven't dug too deep in this yet, but it looks like Keycloak 11 is expecting `groups.path` for the LDAP group mapper to always be provided, and all previous versions are failing when that is provided.

If this is the case, then we'll have to start doing version checking at runtime instead of just doing it during tests.  I can dig into this more when I have some free time.

**Comment by klausenbusk** [comment]
>  I can dig into this more when I have some free time.

I would appreciate that! I just experimented a little in the "wip" commit, but it still lacks a `if version < 11.0.0 set ldap_groups_path = default`, and I'm not sure it is implemented the best way. Maybe you have a better idea/design in mind :)

**Comment by klausenbusk** [comment]
@mrparkers Could you take a look? I just added `Computed: true` which is a much simpler solution.

**Comment by klausenbusk** [comment]
> Hmm... Now 11.0.0 is broke..

Now all the CI is green. Could you take a look at the last commit?

File: keycloak/ldap_group_mapper.go [hunk 1]
```diff
@@ -69,9 +69,10 @@ func convertFromLdapGroupMapperToComponent(ldapGroupMapper *LdapGroupMapper) *co
 		"drop.non.existing.groups.during.sync": {
 			strconv.FormatBool(ldapGroupMapper.DropNonExistingGroupsDuringSync),
 		},
-		"groups.path": {
-			ldapGroupMapper.LdapGroupsPath,
-		},
+	}
+
+	if KeycloakVersionIsGreaterThanOrEqualTo(keycloakClient, "11.0.0") {
```
Comment (klausenbusk) [inline keycloak/ldap_group_mapper.go:74]: I don't think we need this, but maybe it worth it anyway as it is more explicit.

**Comment by svenstaro** [comment]
@klausenbusk This needs a rebase.

**Comment by klausenbusk** [comment]
I think this was <somehow> fixed as part of: https://github.com/mrparkers/terraform-provider-keycloak/pull/369

**Pull request closed without merging**